cinema/views.py

```diff
@@ -48,6 +75,35 @@ class MovieSessionViewSet(viewsets.ModelViewSet):
     queryset = MovieSession.objects.all()
     serializer_class = MovieSessionSerializer
 
+    @staticmethod
+    def _params_to_ints(qs):
+        return [int(str_id) for str_id in qs.split(",")]
+
+    def get_queryset(self):
+        queryset = self.queryset
+
+        date = self.request.query_params.get("date")
+        movie = self.request.query_params.get("movie")
+
+        if self.action == "list":
+            queryset = (
+                queryset
+                .select_related("movie", "cinema_hall")
+                .annotate(tickets_available=(
+                    F("cinema_hall__rows") * F("cinema_hall__seats_in_row")
+                    - Count("tickets"))
+                )
+            )
+            if date:
+                target_date = datetime.strptime(date, "%Y-%m-%d")
+                queryset = queryset.filter(show_time__date=target_date)
+            if movie:
+                movie_ids = self._params_to_ints(movie)
+                queryset = queryset.filter(movie__id__in=movie_ids)
```

Comment:
There is no need for double underscores here for `movie__id`, use `movie_id` instead.